### ISSUE TITLE
integrate Latex rendering into quotebot

### ISSUE THREAD

**Issue by synsypa** [zhangchuck/discord_quote_bot#29]
We don't actually use any of the extra features in MathBot. might be worthwhile just to integrate Latex rendering into base quotebot and ditch mathbot

**Comment by zhangchuck** [comment]
@KingXiaoDi @DoubleSec What would the preferred interface/user experience be?

- Alternative 1: Works like mathbot
    - Alice sends `!latex e^{2}` or `$$ e^{2}`
    - Mathbot posts an image of the formula and posts as mathbot
- Alternative 2: Allow discord_quote to edit Alice's post
    - Alice sends `!latex e^{2}` or `$$ e^{2}`
    - Mathbot generates the image, and attaches it to Alice's original message
    - Mathbot edits the original message to empty
- Alternative 2: Allow discord_quote to edit Alice's post and allow it to be samless
    - Alice sends `!latex I'm really interested in $e^{2}$`
    - Mathbot detects that $e^{2}$ is a valid latex math expression 
    - Mathbot generates the image for $e^{2}$, and attaches it to Alice's original message
    - Mathbot edits the original message by removing `!latex` and replaceing `$..$` with `[EXPRESSION]`
    - Note, for simplicity, we would want to limit each message to a single math expression

**Comment by synsypa** [comment]
in general my preference would be to post as mathbot. i think we walk into iffy territory letting the bot do too much editing of the original message

**Comment by DoubleSec** [comment]
I agree, probably better to just have it post a new message. Do we want to have it delete the request like it does for `!quote`?

**Comment by zhangchuck** [comment]
@synsypa I think Mathbot had an issue where the request and the image got a bit divorced, because it took time to move the image back and forth from the bot.

@DoubleSec Maybe? But then you lose any non-latex context in the message. Maybe it should just look like/function as a quote? e.g., imagine `Alternative 2B` but instead of editing the message, quote-bot says the message
